hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -796,6 +855,7 @@ public String toString() {
                 + ", mapAttributeConfigs=" + mapAttributeConfigs
                 + ", quorumName=" + quorumName
                 + ", queryCacheConfigs=" + queryCacheConfigs
+                + ", cacheDeserializedValues=" + cacheDeserializedValues
                 + '}';
```

Comment:
@jerrinot i do not have a reason to not to add `cacheDeserializedValues ` into hashCode and equals methods. Seems `optimizeQuery` is a missed one also like `hotRestartEnabled`. 
They can be added.